cabra/src/pages/components/CreateTaskForm.tsx

```diff
@@ -16,13 +15,22 @@ const emptyTask: Omit<ITask, "id"> = {
 
 export default function CreateTaskForm() {
   const navigate = useNavigate();
+  const [params] = useSearchParams();
   const client = useQueryClient();
   const addTask = useMutation(createTask, {
     onSuccess: () => {
       client.invalidateQueries(getTasks.cacheKey());
-      navigate(routeHelpers.tasks);
+      navigate(-1);
     },
   });
 
-  return <TaskForm onSubmit={addTask.mutateAsync} task={emptyTask} />;
+  return (
+    <TaskForm
+      onSubmit={addTask.mutateAsync}
+      task={{
+        ...emptyTask,
+        dueDate: params.get("date"),
```

Comment:
Why do you pass it here using query params, instead of using props?

Reply:
It would be hard to pass it way down from the calendar event to here and it would disappear after refreshing the page. When it's done with query params you can go to the URL with the date param any time, and it will always provide the default date in the form.